### ISSUE TITLE
Changelog compares wrong versions

### ISSUE THREAD

**Issue by AntJanus** [conventional-changelog/standard-version#126]
I ran my first release a few days ago and bumped my original `2.0.0` to `2.4.0` which is okay (I don't mind skipping 3 versions) and it generated a neat changelog. When I ran `standard-version` again today, it compared 2.5.0 against 2.0.0 instead of 2.4.0

Not sure why it happened but `package.json` has the right version and there is a tag/release for `2.4.0` as well. Any idea on why it would skip?

Basically `2.5.0` was a duplicate of `2.4.0` with a couple of extra features.

As a sitenote, I did have an existing changelog so could it be a parsing issue?

**Comment by Tapppi** [comment]
IIRC the last version will be fetched from the `git tag -l` list or the git history (@stevemao ?).. @bcoe didn't you have similar issues previously? IIRC the problem was with the tags leading to a dead branch and the history being inconsistent. Could you check that the `v2.4.0` tag points to a commit in your `master` (or whatever your release branch is) branches history @AntJanus ?

**Comment by bcoe** [comment]
@Tapppi I think I described a similar issue again here: https://github.com/conventional-changelog/standard-version/issues/123

**Comment by Tapppi** [comment]
Closing due to inactivity and I'm fairly certain this is covered by the workflow problems in #123 and https://github.com/conventional-changelog/git-semver-tags/pull/10.

**Comment by AntJanus** [comment]
@Tapppi I'm horrible at coming back to issues I file! But yes, this was resolved for me when I started to run `standard-version` on my main branch rather than branching off to create the tag.
